### PR TITLE
add level-js5 upgrade code

### PR DESCRIPTION
**Pull request by tabcat**
handles upgrading level-js stores that have yet to be upgraded per https://github.com/Level/level-js/blob/master/UPGRADING.md#500

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -57,14 +57,49 @@ class Storage {
...
+  // upgrade level-js to version 5.0.0
+  // https://github.com/Level/level-js/blob/master/UPGRADING.md#500
+  async _leveljsUpgrade (store, leveljs) {
+    const upgradeKey = new Uint8Array([0])
+
+    async function isUpgraded () {
+      return Boolean(await store.get(upgradeKey).catch(e => false))
+    }
+    async function setUpgraded () {
+      await store.put(upgradeKey, '1')
```
Comment (aphelionz) [inline src/index.js:86]: Should we keep the version that we've upgraded to here i.e. `5` instead of the magic number `1`?

Comment (tabcat) [reply src/index.js:86]: that sounds like a good idea :+1: these stores will only ever be written to using level-js as well so level-js version 5 is implied

File: src/index.js [hunk 2]
```diff
@@ -57,14 +57,49 @@ class Storage {
...
+    }
+
+    // return if already upgraded
+    if (await isUpgraded()) {
+      store._leveljs5 = true
```
Comment (aphelionz) [inline src/index.js:91]: let's change the tests to read from the db directly rather than this stray object property

**Review by aphelionz — CHANGES_REQUESTED** [review]

**Comment by tabcat** [comment]
i think we should hold this back for release just yet.

somethings to do:
- add to the level-js upgrade test to make sure [the upgrade is skipped if already upgraded](https://github.com/orbitdb/orbit-db-storage-adapter/pull/8/files#diff-7b958a0f7650d8b6384de4170554519fe6cca202d21414b05fe1782c5bd5ea6dR14)
- do more upgrade testing in the orbit-db repo with more real data to see how much time it takes; and whether we can upgrade portions while reading from non upgraded portions with the new version of level-js.

**Pull request closed without merging**